airflow/example_dags/example_kubernetes_executor_config.py

```diff
@@ -52,60 +53,67 @@ def test_volume_mount():
     start_task = PythonOperator(
         task_id="start_task",
         python_callable=print_stuff,
-        executor_config={
-            "KubernetesExecutor": {
```

Comment:
I think having the executor name would be helpful if in future we want to see configs for multiple executors

Reply:
The problem I'm seeing is that k8s.V1Pod objects aren't serializable into dictionaries. Not sure how to get around that and would rather not optimize for a future I don't think will come

Reply:
(at this point the executor_config has existed for years and no other executor has used it. If it wasn't such a pain I'd push for us to change the name to something like "kubernetes_modifier_pod")

Reply:
> The problem I'm seeing is that k8s.V1Pod objects aren't serializable into dictionaries. Not sure how to get around that and would rather not optimize for a future I don't think will come

We will need it for DAG Serialization too, is it not possible to convert V1Pod to dict ? I thought each k8s object had some kind of method for that

Reply:
Let me try again. Maybe I'm missing something.

Reply:
```  
File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/json.py", line 54, in _default
    raise TypeError(f"Object of type '{obj.__class__.__name__}' is not JSON serializable")
TypeError: Object of type 'V1Pod' is not JSON serializable
```

Yeah so if I try to make the pod a field I end up getting this error when I try to render on the page. Not sure what the best way forward is there. Maybe add a field? I don't think we need to change the DB as the executor_config isn't stored in the DB anyways.

Reply:
(though the downside of adding a field is that it would create even more confusing in transitioning DAGs. I'd personally either figure out a solution to this json serialization issue, or just have multiple types for the executor_config)

Reply:
We do store executor_config in TaskInstance table:

https://github.com/apache/airflow/blob/d7602654526fdd2876466371404784bd17cfe0d2/airflow/models/taskinstance.py#L222

Although we use `dill` to store it

Reply:
`k8s.V1Pod(...).to_dict` should work for serialization, isn't it?

https://github.com/kubernetes-client/python/blob/v11.0.0/kubernetes/client/models/v1_pod.py#L179-L201

Reply:
@kaxil yeah but forcing using to use the to_dict function seems kinda clunky, no?

We store the executor_config in the DB but TBH we don't really need to. Maybe we can just create new fields "kubernetes_executor_config" and "dag_serialization_config" and just remove the "executor_config" as a whole?

Reply:
No, I meant, adding code in serialization module, https://github.com/apache/airflow/blob/e565368f2e988a06b0398b77e66859c138905ae7/airflow/serialization/serialized_objects.py#L159-L216

that if the object is k8s.V1Pod, store it as `k8s.V1Pod.to_dict()` .

Reply:
>We store the executor_config in the DB but TBH we don't really need to. Maybe we can just create new fields "kubernetes_executor_config" and "dag_serialization_config" and just remove the "executor_config" as a whole?

If we don't need to store it in DB then sure we can remove it

Reply:
Ok I got it working with the traditional executor_config. Thanks @kaxil!